test/services/homework/index.test.js

```diff
@@ -252,196 +278,242 @@ describe('homework service', function test() {
+			expect(result.data[0].stats.submissionPercentage).to.equal('50.00');
+			expect(result.data[0].stats.gradeCount).to.equal(1);
+			expect(result.data[0].stats.gradePercentage).to.equal('50.00');
+			expect(result.data[0].stats.averageGrade).to.equal('67.00');
+			// no grade as a teacher
+			expect(result.data[0].grade).to.equal(undefined);
+		});
 
-	it('contains statistics as students when publicSubmissions:true', async () => {
-		const { students, homework } = await setupHomeworkWithGrades();
-		await homeworkService.patch(homework._id, { publicSubmissions: true });
-		const params = await testObjects.generateRequestParamsFromUser(students[0]);
-		params.query = { _id: homework._id };
-		const result = await homeworkService.find(params);
-		expect(result.data[0].grade).to.equal('67.00');
-		// no stats as a student
-		expect(result.data[0].stats).to.not.equal(undefined);
-	});
+		it('student sees grade but no stats', async () => {
+			const { students, homework } = await setupHomeworkWithGrades();
 
-	it('teacher can PATCH his own homework', async () => {
-		const [teacher] = await Promise.all([testObjects.createTestUser({ roles: ['teacher'] })]);
-		const course = await testObjects.createTestCourse({
-			teacherIds: [teacher._id],
-			userIds: [],
+			const params = await testObjects.generateRequestParamsFromUser(students[0]);
+			params.query = { _id: homework._id };
+			const result = await homeworkService.find(params);
+			expect(result.data[0].grade).to.equal('67.00');
+			// no stats as a student
+			expect(result.data[0].stats).to.equal(undefined);
 		});
-		const homework = await testObjects.createTestHomework({
-			teacherId: teacher._id,
-			name: 'Testaufgabe',
-			description: '\u003cp\u003eAufgabenbeschreibung\u003c/p\u003e\r\n',
-			availableDate: Date.now(),
-			dueDate: '2030-11-16T12:47:00.000Z',
-			private: false,
-			archived: [teacher._id],
-			lessonId: null,
-			courseId: course._id,
+
+		it('student sees if he submitted or not', async () => {
+			const { students, course, homework } = await setupHomeworkWithGrades();
+			const submissionTwo = await testObjects.createTestSubmission({
+				schoolId: course.schoolId,
+				courseId: course._id,
+				homeworkId: homework._id,
+				studentId: students[1]._id,
+				comment: 'hello teacher, his dog has eaten this database entry also...',
+				grade: 67,
+			});
+
+			const params = await testObjects.generateRequestParamsFromUser(students[0]);
+			params.query = { _id: homework._id };
+			const result = await homeworkService.find(params);
+			expect(result.data[0].submissions).to.equal(1);
+		});
+
+		it('homework contains course details', async () => {
+			const { teacher, homework } = await setUpHOmework();
+			const params = await testObjects.generateRequestParamsFromUser(teacher);
+			params.query = {};
+			const result = await homeworkService.find(params);
+			expect(result.data[0].schoolId.toString() === result.data[0].courseId.schoolId.toString());
```

Comment:
do you miss here to check if it is true? Would also be better to make expect statements simple for a more simple understanding